lib/src/results_storage.cc

```diff
@@ -186,4 +191,55 @@ ActionOutput ResultsStorage::getOutput(const std::string& transaction_id,
     return output;
 }
 
+static void defaultPurgeCallback(const std::string& dir_path)
+{
+    fs::remove_all(dir_path);
+}
+
+unsigned int ResultsStorage::purge(
+                const std::string& ttl,
+                const std::vector<std::string>& ongoing_transactions,
+                std::function<void(const std::string& dir_path)> purge_callback)
+{
+    int num_purged_dirs { 0 };
+    Timestamp ts { ttl };
+    if (purge_callback == nullptr)
+        purge_callback = &defaultPurgeCallback;
+
+    lth_file::each_subdirectory(
+        spool_dir_path_.string(),
+        [&](std::string const& s) -> bool {
+            fs::path dir_path { s };
+            auto transaction_id = dir_path.filename().string();
+
+            if (!ongoing_transactions.empty()
+                    && std::find(ongoing_transactions.begin(),
+                                 ongoing_transactions.end(),
+                                 transaction_id) != ongoing_transactions.end())
+                return true;
+
+            try {
+                auto md = getActionMetadata(transaction_id);
+                if (md.get<std::string>("status") != "running"
+                        && !ts.isOlderThan(md.get<std::string>("start"))) {
```

Comment:
Mine too, but when I actually used Timestamp and got that feeling, my lazy side prevailed...